resources.yaml

```diff
@@ -180,6 +180,170 @@ items:
+              dnsPolicy: ClusterFirst
+              restartPolicy: Always
+              serviceAccount: oshinko
+          triggers:
+            - type: ConfigChange
+            - type: ImageChange
+              imageChangeParams:
+                automatic: true
+                containerNames:
+                  - ${APPLICATION_NAME}
+                from:
+                  kind: ImageStreamTag
+                  name: ${APPLICATION_NAME}:latest
+      - apiVersion: v1
+        kind: Service
+        metadata:
+          name: ${APPLICATION_NAME}
+          labels:
+            app: ${APPLICATION_NAME}
+        spec:
+          ports:
+            - name: 8080-tcp
+              port: 8080
+              protocol: TCP
+              targetPort: 8080
+          selector:
+            deploymentconfig: ${APPLICATION_NAME}
+    parameters:
+      - name: APPLICATION_NAME
+        displayName: Application Name
+        description: The name to use for the BuildConfig, ImageStream and DeploymentConfig components
+        from: pyspark-[a-z0-9]{4}
+        generate: expression
+      - name: GIT_URI
+        displayName: Git Repository URL
+        description: The URL of the repository with your application source code
+        required: true
+      - name: APP_ARGS
+        displayName: Application Arguments
+        description: Command line arguments to pass to the application
+      - name: SPARK_OPTIONS
+        displayName: spark-submit Options
+        description: List of additional options to pass to spark-submit (for exmaple --conf property=value or --package ...). --master and --class are set by the launcher and should not be set here.
+      - name: GIT_REF
+        displayName: Git Reference
+        description: Optional branch, tag or commit
+      - name: OSHINKO_CLUSTER_NAME
+        description: The name of the spark cluster to run against. The cluster will be created if it does not exist, and a random cluster name will be chosen if this value is left blank.
+      - name: OSHINKO_NAMED_CONFIG
+        description: The name of a stored cluster configuration to use if a cluster is created, default is 'default'.
+      - name: OSHINKO_SPARK_DRIVER_CONFIG
+        description: The name of a configmap to use for the spark configuration of the driver. If this configmap is empty the default spark configuration will be used.
+      - name: OSHINKO_DEL_CLUSTER
+        description: If a cluster is created on-demand, delete the cluster when the application finishes if this option is set to 'true'
+        value: "true"
+        required: true
+      - name: APP_FILE
+        description: The name of the main JAR file. If this is not specified and there is a single JAR produced by the build, that JAR will be chosen.
+      - name: APP_EXIT
+        description: Setting this value to 'false' prevents the application from being re-deployed if/when it completes
+        value: "false"
+        required: true
+      - name: APP_MAIN_CLASS
```

Comment:
for a java app the main class must be high priority and not guessable. it should be higher in the list (will display higher in the console) and have a displayName.